libcontainer/state_linux.go

```diff
@@ -38,7 +38,8 @@ type containerState interface {
 }
 
 func destroy(c *linuxContainer) error {
-	if !c.config.Namespaces.Contains(configs.NEWPID) {
+	if !c.config.Namespaces.Contains(configs.NEWPID) ||
+		c.config.Namespaces.PathOf(configs.NEWPID) != "" {
```

Comment:
Killing PID 1 in a PID namespace kills the other processes in the PID namespace.

Reply:
signalAllProcesses only kill the processes found from the container's cgroup, when a container share pid ns, it will not kill the pid 1 process in this pid ns.